cmake/SundialsBuildOptionsPre.cmake

```diff
@@ -93,10 +93,15 @@ sundials_option(SUNDIALS_LOGGING_ENABLE_MPI BOOL "${DOCSTR}" "${ENABLE_MPI}"
 # ---------------------------------------------------------------
 
 if(UNIX)
-  sundials_option(USE_GENERIC_MATH BOOL "Use generic (std-c) math libraries" ON)
-  # all executables will be linked against -lm
-  set(EXTRA_LINK_LIBS -lm)
+  sundials_option(USE_GENERIC_MATH BOOL "Use the C standard library math functions" ON ADVANCED)
+  if(USE_GENERIC_MATH)
+    sundials_option(SUNDIALS_MATH_LIBRARY PATH "Which math library (e.g., libm) to link to" "-lm" ADVANCED)
+  endif()
+else()
+  sundials_option(SUNDIALS_MATH_LIBRARY PATH "Which math library (e.g., libm) to link to" "" ADVANCED)
```

Comment:
Add `set(USE_GENERIC_MATH ON)` in the non-Unix case so that it is defined for later checks.